tests: Add checkpoint tests for all ISAs

Harshil2107 opened the pull request:
Added save and restore checkpoint tests for arm-hello, x86-hello, x86-fs, power-hello 
Added mips and sparc test but mips does not support checkpoint and there is a bug in sparc. 
Added test file to run the tests.

Change-Id: I2d3b96f95ee08aae921de9a885ac5be77d49f326

Harshil2107 committed:
tests: Add checkpoint tests for all ISAs

Added save and restore checkpoint tests for arm-hello, x86-hello, x86-fs, power-hello
Added mips and sparc test but mips does not support checkpoint and there is a bug in sparc.
Added test file to run the tests.

Change-Id: I2d3b96f95ee08aae921de9a885ac5be77d49f326

BobbyRBruce commented on tests/gem5/checkpoint-tests/arm-hello-restore-checkpoint.py:
```diff
@@ -0,0 +1,80 @@
+# Copyright (c) 2022 The Regents of the University of California
```
The copyright should be 2023. Update this for all relevant files

BobbyRBruce commented on tests/gem5/checkpoint-tests/mips-hello-restore-checkpoint.py:
```diff
@@ -0,0 +1,72 @@
+"""
+This gem5 configuation script creates a simple board sharing the same
+structure as the one in
+tests/gem5/checkpoint-tests/mips-hello-save-checkpoint.py.
```
There's no way you could've known this, but all directories in the "tests" directory should be snakecase, e.g. `checkpoint_tests` -- all lower case with underscores as spaces. The python files should be with dashes `mips-hello-save-checkpoint.y` (so keep them as they are).

The reason for this is that python can't import paths containing dashes (.e.g., `import checkpoint-tests.script` will fail). So in a python project it's best the directories camel case. For the python files themselves, you use dashes for the executable script, and underscores in module names which are supposed to be imported. This is a neat way to avoid people importing scripts designed to be directly executed.

Melissa recently pushed a patch which renamed all the directories in tests to conform to this standard.

BobbyRBruce commented on tests/gem5/checkpoint-tests/arm-hello-restore-checkpoint.py:
So as of #156 we're doing the following directory structure:

```txt
tests/
   gem5/
      checkpoint_tests
         README.md # A md file explaining the purpose of the tests
         configs # a directory containing all the configuration scripts
              arm-hello-restore-checkpoint-.py
              ...
         test-checkpoints.py
```

Please refactor this. Remember to update all the paths you used.

BobbyRBruce commented on tests/gem5/checkpoint-tests/test-checkpoints.py:
```diff
@@ -0,0 +1,266 @@
+# Copyright (c) 2021-2023 The Regents of the University of California
```
Why is this 2023 to 2023?. This is a new file, no?

BobbyRBruce commented on tests/gem5/checkpoint-tests/test-checkpoints.py:
```diff
@@ -0,0 +1,266 @@
+    length=constants.quick_tag,
+)
+
+# gem5_verify_config(
```
Add a link here to the issue you created/will create to not this bug (also link the issue back to this PR).

Harshil2107 replied:
Should I remove the mips and sparc test files?

BobbyRBruce replied:
Why would you remove those?

Harshil2107 replied:
I thought of removing mips because it does not support checkpoints. I don't know if the support will be added.

BobbyRBruce replied:
If mips genuinely doesn't support checkpoints then you're fine to remove these tests. I want tests there for stuff that is supposed to be supported but is broken.

powerjg replied:
+1 to remove MIPS

Harshil2107 replied:
Just removed the mips test, I will add the link to the issue when I make one.

BobbyRBruce reviewed (changes requested):
Good work! I've got some things that need changed but it's all quite minor.

Harshil2107 committed:
tests: Updated directory structure

-Changed copyright message to reflect correct year.
- Updated directory structure.
- Changed directory name to snake case.
- Added a README.md for checkpoint tests.

Change-Id: Id350addb9cce6740a20a5a45171f80306b711efa

Harshil2107 committed:
tests: Removed mips checkpoint tests

Change-Id: I03ad0025ec982245721fd7faad8d75cdbb99cf81

BobbyRBruce commented on tests/gem5/checkpoint_tests/test-checkpoints.py:
```diff
@@ -0,0 +1,278 @@
+    length=constants.quick_tag,
+)
+
+# gem5_verify_config(
```
We've removed MIPS, so remove the call to the tests.

BobbyRBruce commented on tests/gem5/checkpoint_tests/test-checkpoints.py:
```diff
@@ -0,0 +1,278 @@
+#     length=constants.quick_tag,
+# )
+
+# gem5_verify_config(
```
Add an Issue link here to a GitHub Issue page explaining this bug.

Harshil2107 committed:
tests: removed mips tests and added issue link.

- Removed MIPS tests.
- Added link to github issue sparc test bug.

Change-Id: Ib3c69dca578371ecf0ac2d7694f46f24834a7e5f
